### DISCUSSION TITLE
channel connection status remains "POLLING"

### DISCUSSION THREAD

**Discussion by YALouis** [nextgenhealthcare/connect#4618] [Q&A]
I used mirth with version 3.4.2,it runs well at most time,but sometimes the status would remains "POLLING" and could NOT read message from database.Most of the channels with this problem use "Database Reader" as the source connect.When I forced stop the channel which status remains "POLLING" and restart it,that channel could work again.My first language is not English,please ignore my inappropriate words .
![image](https://user-images.githubusercontent.com/35215030/123790259-aff97f00-d910-11eb-989d-9494bfb864be.png)

**Comment by jonbartels** [comment]
MC 3.4.2 is relatively old. Please consider an upgrade.

What does the connection log view show when your MC is stuck in the POLLING state? This is what the connection log looks like:
<img width="1532" alt="image" src="https://user-images.githubusercontent.com/7669292/123974165-6c435b80-d98a-11eb-817b-1c2933171555.png">

How complicated is the SQL query used for polling? It may be slow or stuck polling if the query is slow or returns many results.

**Reply by pacmano1** [reply]
This question is also on the forums.
